client/src/components/R&R/NewReview.jsx

```diff
@@ -4,72 +4,279 @@ import $ from 'jquery';
 import axios from 'axios';
 import styled from 'styled-components';
 import handler from '../Shared/reviewhandler.js';
-
-
+import StarRating from './InteractiveStars.jsx';
+import NewReviewChars from './NewReviewChars.jsx';
+import Errors from './Errors.jsx';
 
 class NewReview extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       show: false,
       stars: null,
-      recommend: null,
-      characteristics: null,
+      recommend: true,
+      characteristics: {},
       summary: '',
       body: '',
       images: [],
       nickname: '',
       email: '',
-      count: 60,
+      count: `Minimum required characters left: 50`,
+      imgWindow: false,
+      submitted: false,
+      errors: {},
+      showError: false
     };
+
+    this.handleInputChange = this.handleInputChange.bind(this);
     this.toggleModal = this.toggleModal.bind(this);
+    this.toggleImgWindow = this.toggleImgWindow.bind(this);
     this.submitReview = this.submitReview.bind(this);
+    this.characterCounter = this.characterCounter.bind(this);
+    this.processCharSelection = this.processCharSelection.bind(this);
+  }
+
+  characterCounter(e) {
+    let charsLeft = 50 - e.target.value.length;
+    if (charsLeft > 0) {
+       this.setState({ count: `Minimum required characters left: ${charsLeft}` });
+    } else {
+      this.setState({ count: 'Minimum reached' });
+    }
+  }
+
+  handleInputChange(e, callback = () => {}) {
+    const name = e.target.getAttribute('name');
+    const value = e.target.value || e.target.getAttribute('value');
+
+    console.log('e.target >>>>', e.target);
+    this.setState({ [name]: value }, () => console.log(`${name} in state is now: `, this.state[name]), callback(e));
+  }
+
+  processCharSelection(name, value) {
+    console.log('this.props.characteristics >>>', this.props.characteristics);
```

Comment:
you have some good debugging logs here. consider a curried function that allows you to turn this on or off based on a global constant `DEBUG`